### DISCUSSION TITLE
Adding routes to group doesn't work

### DISCUSSION THREAD

**Discussion by sammyskills** [codeigniter4/shield#259] [Q&A]
Hello,

I've tried to change the default route by grouping it like so:

```php
$routes->group('account', function($routes) {
    service('auth')->routes($routes);
});
```

But it doesn't work as expected, visiting `account/login` OR `account/register`, returns a 404 error and when I run `php spark routes` command, it shows:

`account///login`
`account///register`

I understand that from the [docs](https://github.com/codeigniter4/shield/blob/develop/docs/7%20-%20customization.md#route-configuration), it is possible to customize the routes, but what if I just want to put all the routes in a group, without having to do that for all individual routes?

**Comment by kenjis** [comment]
There are route definitions as an array in the Config class:
https://github.com/codeigniter4/shield/blob/develop/src/Config/AuthRoutes.php
You can get them, and you could build route definitions by them.

**Comment by kenjis** [comment]
This patch solves this problem.
I don't know if this problem is a bug in Shield or CI4 routing, but it seems to solve this problem.

```diff
--- a/src/Auth.php
+++ b/src/Auth.php
@@ -102,7 +102,7 @@ public function routes(RouteCollection &$routes, array $config = []): void
     {
         $authRoutes = config('AuthRoutes')->routes;
 
-        $routes->group('/', ['namespace' => 'CodeIgniter\Shield\Controllers'], static function (RouteCollection $routes) use ($authRoutes, $config): void {
+        $routes->group('', ['namespace' => 'CodeIgniter\Shield\Controllers'], static function (RouteCollection $routes) use ($authRoutes, $config): void {
             foreach ($authRoutes as $name => $row) {
                 if (! isset($config['except']) || (isset($config['except']) && ! in_array($name, $config['except'], true))) {
                     foreach ($row as $params) {
```

**Reply by kenjis** [reply]
I sent a PR to fix: https://github.com/codeigniter4/CodeIgniter4/pull/6186

**Reply by kenjis** [reply]
@sammyskills This bug has been fixed in the `develop` branch of CodeIgniter4, and it will be included in v4.2.2.

**Reply by sammyskills** [reply]
Thanks @kenjis.

**Reply by sanjeevdivekar** [reply]
This is my Routes.php code
```
$routes->group('admin', function ($routes) {
    service('auth')->routes($routes);
});
```
Still, I am facing an error.

![image](https://user-images.githubusercontent.com/1086717/178147695-afe2378b-bf39-45f1-a9c9-b293f72db99b.png)

Any suggestions?

**Reply by kenjis** [reply]
@sanjeevdivekar Do you use CI4 of `develop` branch?
v4.2.2 is not yet released.

**Reply by sanjeevdivekar** [reply]
No. I am using v4.2.1

**Reply by kenjis** [reply]
You cannot use in v4.2.1:
```php
$routes->group('admin', function ($routes) {
    service('auth')->routes($routes);
});
```
Wait for v4.2.2 or use `develop` branch.

**Reply by sammyskills** [reply]
Just in case someone stumbles on this before the release of CI-4.2.2, you can customize _(exempt)_ the routes before adding them in a group like so:

```
service('auth')->routes($routes, ['except' => ['login', 'register', 'magic-link']]);
$routes->group('admin', static function ($routes) { 
    $routes->get('login', '\CodeIgniter\Shield\Controllers\LoginController::loginView', ['as' => 'login']);
    $routes->post('/', '\CodeIgniter\Shield\Controllers\LoginController::loginAction');
    $routes->get('register', '\CodeIgniter\Shield\Controllers\LoginController::loginView', ['as' => 'register']);
    $routes->post('/', '\CodeIgniter\Shield\Controllers\LoginController::loginAction');
    $routes->get('magic-link', '\CodeIgniter\Shield\Controllers\MagicLinkController::loginView', ['as' => 'magic-link']);
    $routes->post('magic-link', '\CodeIgniter\Shield\Controllers\MagicLinkController::loginAction');
    $routes->get('verify-magic-link', '\CodeIgniter\Shield\Controllers\MagicLinkController::verify', ['as' => 'verify-magic-link']);
});
```

**Reply by datamweb** [reply]
Shield currently requires version **CI4.2.7.** See https://github.com/datamweb/shield#prerequisite.

**Reply by sammyskills** [reply]
Oh, yes, you are correct.

Unfortunately, routing to named routes (login and register) within the group returns an error. This is because they weren't declared in the `Config/AuthRoutes.php` file. 

I'll submit a PR now to fix that.

**Reply by sammyskills** [reply]
PR #577 sent.
